File: app/src/main/java/com/komsi/maleshop/adapter/SlideshowAdapter.java
package com.komsi.maleshop.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.viewpager.widget.PagerAdapter;

import com.bumptech.glide.Glide;
import com.komsi.maleshop.R;
import com.komsi.maleshop.model.Slide;

import java.util.List;

public class SlideshowAdapter extends PagerAdapter {
   private Context mcontext;
   public List<Slide> mList;


    public SlideshowAdapter(Context mcontext, List<Slide> mList) {
        this.mcontext = mcontext;
        this.mList = mList;
    }

    @Override
    public int getCount() {
        return mList.size();
    }

    @NonNull
    @Override
    public Object instantiateItem(@NonNull ViewGroup container, int position) {
        LayoutInflater inflater = (LayoutInflater) mcontext.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        View slideLayout = inflater.inflate(R.layout.slider_item,null);

        ImageView slideImg = slideLayout.findViewById(R.id.slide_img);
        TextView slideText = slideLayout.findViewById(R.id.slide_title);
        Glide.with(mcontext).load(mList.get(position).getImage()).into(slideImg);
        slideText.setText(mList.get(position).getTitle());

        container.addView(slideLayout);
        return slideLayout;



    }

    @Override
    public boolean isViewFromObject(@NonNull View view, @NonNull Object o) {
        return view==o;
    }

    @Override
    public void destroyItem(@NonNull ViewGroup container, int position, @NonNull Object object) {
        container.removeView((View)object);
    }
}

File: app/src/main/java/com/komsi/maleshop/adapter/SlideshowAdapter.kt
package com.komsi.maleshop.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.viewpager.widget.PagerAdapter
import com.bumptech.glide.Glide
import com.komsi.maleshop.R
import com.komsi.maleshop.model.Slide
import java.util.*

class SlideshowAdapter(private val mcontext: Context) : PagerAdapter() {
    var mList = ArrayList<Slide>()
    fun setData(list: ArrayList<Slide>) {
        mList.clear()
        mList.addAll(list)
        notifyDataSetChanged()
    }

    override fun getCount(): Int {
        return mList.size
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val inflater = mcontext.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        val slideLayout = inflater.inflate(R.layout.slider_item, null)
        val slideImg = slideLayout.findViewById<ImageView>(R.id.slide_img)
        val slideText = slideLayout.findViewById<TextView>(R.id.slide_title)
        Glide.with(mcontext).load(mList[position].image).into(slideImg)
        slideText.text = mList[position].title
        container.addView(slideLayout)
        return slideLayout
    }

    override fun isViewFromObject(view: View, o: Any): Boolean {
        return view === o
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        container.removeView(`object` as View)
    }

}